lib/matplotlib/dates.py

```diff
@@ -1254,6 +1241,12 @@ def __init__(self, bymonthday=None, interval=1, tz=None):
 
         Default is to tick every day of the month: ``bymonthday=range(1,32)``
         """
+        if interval < 1:
+            raise ValueError("The interval parameter must be an integer "
```

Comment:
since these are distinct errors, the messages should be unique:
interval < 1 : The interval must be greater than 1
isinstance(interval, int): the interval must be an integer
Otherwise you may as well do `if interval <1 or not isinstance(interval, int)`